### ISSUE TITLE
Object not found when using --commit flag

### ISSUE THREAD

**Issue by asaini11** [zricethezav/gitleaks#686]
**Describe the bug**
We are running gitleaks as a pre-receive hook on GitLab using `gitleaks --leaks-exit-code=2 --path=. --verbose --commit=$commit` where $commit is the commit sha to be checked at the time of pushing. In some cases, gitleaks is unable to run and we get the below 'object not found' with exit code 1. I believe it is an issue with the ` --commit ` option.
```
INFO[0000] opening .                                    
ERRO[0000] object not found
```
We have tried this with and without the `--path` option but no change.

**To Reproduce**
I have tried to run gitleaks locally by running `gitleaks --leaks-exit-code=2 --path=. --verbose --commit=$commit` where the commit is a commit sha already in GitLab and we can see the same error locally. Note. Gitleaks is able to run on some commits in the repo but not others. 

**Expected behavior**
I am expecting gitleaks to run successfully with either `level=info msg="No leaks found"` (exit code 0) or `level=warning msg="leaks found: x"` where x is the number of leaks found with exit code 2 in our case.

**Basic Info (please complete the following information):**
 - Gitleaks Version: We are using the latest version of v.7.6.1.

**Additional info:**
We have tried gitleaks v7.5.0 locally too but get the same issue.

cc @zricethezav

**Comment by w0rmr1d3r** [comment]
Hello @asaini11 

I see this `Note. Gitleaks is able to run on some commits in the repo but not others.` 
I might need more info on that, why on some commits gitleaks fails and why on other commits it runs without any issue.

Do you have more info on why is that?
What are the differences between those commits?

Thank you

**Comment by asaini11** [comment]
Hi @w0rmr1d3r thank you for your comment. Currently, it's unclear what the differences are, as both commits update .java files. I tried to replicate the issue this morning and it replicated but I tried again this afternoon and now can't I replicate the issue as it's scanning. It seems pretty random. However, I wanted to use the `--debug` option to get more information but nothing extra prints out. Is there a way I could try and capture how commits with the 'object not found` error differ to scanned commits using the --debug flag?
